Core/src/org/sleuthkit/autopsy/contentviewers/ApplicationContentViewer.java

```diff
@@ -0,0 +1,251 @@
+                return;
+            } catch (TskCoreException ex) {
+                LOGGER.log(Level.SEVERE, String.format("Could not detect format using fileTypeDetector for file: %s", file), ex); //NON-NLS
+                return;
+            } 
+        }
+        
+        if (mimeType.equalsIgnoreCase("application/octet-stream")) {
+            return;
+        } 
+        else {
+            ContentViewer viewer = getSupportingViewer(mimeType);
+            if (viewer != null) {
+                lastViewer = viewer;
+
+                viewer.setFile(file);
+                this.removeAll();
+                this.add(viewer.getComponent());
+                this.repaint();
+            }
+        }
+       
+    }
+
+    @Override
+    @NbBundle.Messages("ApplicationContentViewer.title=Application Content Viewer")
+    public String getTitle() {
+        return Bundle.ApplicationContentViewer_title();
+    }
+
+    @Override
+    @NbBundle.Messages("ApplicationContentViewer.toolTip=Displays file contents.")
+    public String getToolTip() {
+        return Bundle.ApplicationContentViewer_toolTip();
+    }
+
+    @Override
+    public DataContentViewer createInstance() {
+        return new ApplicationContentViewer();
+    }
+
+    @Override
+    public Component getComponent() {
+        return this;
+    }
+
+    @Override
+    public void resetComponent() {
+
+        if (lastViewer != null) {
+            lastViewer.resetComponent();
+        }
+        this.removeAll();
+        lastViewer = null;
+    }
+
+    @Override
+    public boolean isSupported(Node node) {
+
+        if (node == null) {
+            return false;
+        }
+
+        AbstractFile aFile = node.getLookup().lookup(AbstractFile.class);
+        if (aFile == null) {
+            return false;
+        }
+
+        String mimeType = aFile.getMIMEType();
+        if (Strings.isNullOrEmpty(mimeType)) {
+            LOGGER.log(Level.INFO, "Mimetype not known for file: {0}", aFile.getName()); //NON-NLS
+            try {
+                FileTypeDetector fileTypeDetector = new FileTypeDetector();
+                mimeType = fileTypeDetector.detectFileType(aFile);
+            } catch (FileTypeDetector.FileTypeDetectorInitException ex) {
+                LOGGER.log(Level.SEVERE, "Failed to initialize FileTypeDetector.", ex); //NON-NLS
+                return false;
+            } catch (TskCoreException ex) {
+                LOGGER.log(Level.SEVERE, String.format("Could not detect format using fileTypeDetector for file: %s", aFile), ex); //NON-NLS
+                return false;
+            } 
+        }
+        
+        if (mimeType.equalsIgnoreCase("application/octet-stream")) {
+            return false;
+        } else {
+            return (getSupportingViewer(mimeType) != null);
+        }
+       
+    }
+
+    @Override
+    public int isPreferred(Node node) {
+        if (node == null) {
```

Comment:
isPreferred() is called only if isSupported() returned true.  So, these checks can be dropped.